src/light_data/light_store.rs

```diff
@@ -198,24 +296,8 @@ pub enum LightExpr<F: LurkField> {
         )
     )]
     Comm(F, ScalarPtr<F>),
-    /// Analogous to ScalarExpression::Num
-    #[cfg_attr(
-        not(target_arch = "wasm32"),
-        proptest(strategy = "any::<FWrap<F>>().prop_map(|x| Self::Num(x.0))")
-    )]
-    Num(F),
-    /// Analogous to ScalarExpression::Char
-    #[cfg_attr(
-        not(target_arch = "wasm32"),
-        proptest(strategy = "any::<FWrap<F>>().prop_map(|x| Self::Char(x.0))")
-    )]
-    Char(F),
     /// Analogous to ScalarExpression::Nil
     Nil,
-    /// Analogous to ScalarExpression::Str(""), but as a terminal case of StrCons
-    StrNil,
-    /// Analogous to ScalarExpression::Sym(Sym::root()), but as a terminal case of SymCons
-    SymNil,
 }
```

Comment:
I guess `LightExpr` was not introduced in this PR, but I am commenting here since it's the logical place. I'm a little surprised to discover that `LightExpr` doesn't cover all possible Lurk data. I assume that is because for now it only needs to handle things that are generated by lean.

However, if this is going to be a useful general-purpose serialization format, we do need to handle everything, including keywords, functions, thunks, and continuations (though this is entangled in other conversations, so I mention it more to be exhaustive than to suggest I expected to see it here). cc: @johnchandlerburnham

Reply:
Adding to this, and perhaps to my initial confusion, I note that these missing types *are* handled partially in code above (where my comment on first encountering was to ask, 'why do we not call `create_scalar_ptr()`)?

So I remain a bit confused about the overall handling of these 'missing types'. I do note that these are likely related topics, and that the answer to my question might be 'because we don't actually handle these types (yet)'. However, it's not quite clear to me how it is all meant to fit together.